cmd/crank/beta/lint/cmd.go

```diff
@@ -0,0 +1,203 @@
+type Issue struct {
+	RuleID  string `json:"id"`
+	Name    string `json:"name"`
+	Line    int    `json:"line"`
+	Error   bool   `json:"error"`
+	Message string `json:"message"`
+}
+
+type output struct {
+	Summary struct {
+		Valid    bool `json:"valid"`
+		Total    int  `json:"total"`
+		Errors   int  `json:"errors"`
+		Warnings int  `json:"warnings"`
+	} `json:"summary"`
+	Issues *[]Issue `json:"issues"`
+}
+
+// Run lint.
+func (c *Cmd) Run(k *kong.Context, _ logging.Logger) error {
+	// Load all resources
+	resourceLoader, err := NewLoader(c.Resources)
+	if err != nil {
+		return errors.Wrapf(err, "cannot load resources from %q", c.Resources)
+	}
+
+	resources, err := resourceLoader.Load()
+	if err != nil {
+		return errors.Wrapf(err, "cannot load resources from %q", c.Resources)
+	}
+
+	var allIssues []Issue
+
+	for _, resource := range resources {
+		resourceAsBytes, err := yaml.Marshal(resource)
+		if err != nil {
+			return errors.Wrapf(err, "cannot convert to yaml %q", resource.Value)
+		}
+
+		var k8s unstructured.Unstructured
+		if err := sigyaml.Unmarshal(resourceAsBytes, &k8s); err != nil {
+			return errors.Wrapf(err, "cannot unmarshal into Unstructured %q", resource.Value)
+		}
+
+		name := k8s.GetName()
+
+		if k8s.GetKind() != v1.CompositeResourceDefinitionKind || !strings.HasPrefix(k8s.GetAPIVersion(), v1.Group) {
+			issue := Issue{
+				Name:    name,
+				Line:    0,
+				Error:   true,
+				RuleID:  "XRD000",
+				Message: fmt.Sprintf("Resource %s is not a CompositeResourceDefinition", k8s.GetName()),
+			}
+			allIssues = append(allIssues, issue)
+			continue
+		}
+
+		allIssues = append(allIssues, checkBooleanFields(name, resource)...)
+		allIssues = append(allIssues, checkRequiredFields(name, resource)...)
+		allIssues = append(allIssues, checkMissingDescriptions(name, resource)...)
+	}
+
+	output := output{}
+	output.Summary.Valid = len(allIssues) == 0
+	output.Summary.Total = len(allIssues)
+	output.Issues = &allIssues
+
+	for _, issue := range allIssues {
+		if issue.Error {
+			output.Summary.Errors++
+		} else {
+			output.Summary.Warnings++
+		}
+	}
+
+	if !c.SkipSuccessResults || !output.Summary.Valid {
+		switch c.Output {
+		case "json":
+			err = printJSON(&output, k)
+			if err != nil {
+				return errors.Wrap(err, "cannot print summary")
+			}
+		case "stdout":
+			err = printStdout(&output, k)
+			if err != nil {
+				return errors.Wrap(err, "cannot print summary")
+			}
+		default:
+			return errors.New("invalid output format specified")
+		}
+	}
+
+	if output.Summary.Errors > 0 {
+		os.Exit(1)
+	} else if output.Summary.Warnings > 0 {
+		os.Exit(2)
+	}
+
+	os.Exit(0)
+
+	return nil
+}
+
+func printJSON(o *output, k *kong.Context) error {
+	data, err := json.Marshal(o)
+	if err != nil {
+		return errors.Wrap(err, "cannot marshal output to JSON")
+	}
+	_, err = fmt.Fprintln(k.Stdout, string(data))
+	if err != nil {
+		return errors.Wrap(err, "cannot print summary")
+	}
+	return nil
```

Comment:
Nit: I think you could use `json.NewEncoder(k.Stdout)` here. I don't feel strongly though if you prefer your current approach.

https://pkg.go.dev/encoding/json#Encoder